src/System.Linq/src/System/Linq/Lookup.cs

```diff
@@ -182,7 +196,7 @@ internal List<TResult> ToList<TResult>(Func<TKey, IEnumerable<TElement>, TResult
             return _count;
         }
 
-        public IEnumerable<TResult> ApplyResultSelector<TResult>(Func<TKey, IEnumerable<TElement>, TResult> resultSelector)
+        public IEnumerator<TResult> ApplyResultSelector<TResult>(Func<TKey, IEnumerable<TElement>, TResult> resultSelector)
```

Comment:
Ahh. It would be otherwise a good change. I did not notice it is public. We can't change this.
We can add something like ApplyResultSelectorInternal for internal enumerator use.